File: EsPalindromo.cpp
#include <iostream>

using namespace std;

bool esPalindromo(char cadena[]) {
    int longitud = 0;

    // 1. Calcular la longitud manualmente
    while (cadena[longitud] != '\0') {
        longitud++;
    }

    // 2. Comparar caracteres desde los extremos
    int inicio = 0;
    int fin = longitud - 1;

    while (inicio < fin) {
        // Si los caracteres en las posiciones opuestas no coinciden
        if (cadena[inicio] != cadena[fin]) {
            return false; 
        }
        inicio++;
        fin--;
    }

    return true;
}

int main() {
    char palabra[100];

    cout << "Ingrese una palabra: ";
    cin >> palabra;

    if (esPalindromo(palabra)) {
        cout << "Es un palindromo." << endl;
    } else {
        cout << "No es un palindromo." << endl;
    }

    return 0;
}

File: EsPalindromo.py
def es_palindromo(cadena):
    # 1. Calcular la longitud manualmente
    longitud = 0
    for caracter in cadena:
        longitud += 1
    
    # 2. Comparar desde los extremos hacia el centro
    inicio = 0
    fin = longitud - 1
    
    while inicio < fin:
        # Si encontramos una diferencia, no es palíndromo
        if cadena[inicio] != cadena[fin]:
            return False
        
        inicio += 1
        fin -= 1
        
    return True

# Prueba del algoritmo
palabra = input("Ingrese una palabra o frase: ")

# Limpieza básica para ignorar espacios y mayúsculas (opcional)
# palabra = palabra.replace(" ", "").lower() 

if es_palindromo(palabra):
    print("Es un palíndromo.")
else:
    print("No es un palíndromo.")
